### PR TITLE
Compiler error fixes

### PR DESCRIPTION
**Pull request by PauloCampana**
progress #20505 and #21094

std.valgrind and std.os.windows fixes taken from #21271 and #21272, the author closed their PRs.

### CODE REVIEW

File: lib/std/c.zig [hunk 1]
```diff
@@ -9579,7 +9579,6 @@ pub const system_info = haiku.system_info;
 pub const team_id = haiku.team_id;
 pub const team_info = haiku.team_info;
 pub const thread_id = haiku.thread_id;
-pub const vregs = haiku.vregs;
```
Comment (PauloCampana) [inline lib/std/c.zig:9582]: renamed and is now part of mcontext_t

File: lib/std/fs/File.zig [hunk 2]
```diff
@@ -1691,7 +1691,7 @@ pub fn tryLock(file: File, l: Lock) LockError!bool {
     if (is_windows) {
         var io_status_block: windows.IO_STATUS_BLOCK = undefined;
         const exclusive = switch (l) {
-            .none => return,
+            .none => return true,
```
Comment (alexrp) [inline lib/std/fs/File.zig:1694]:
```suggestion
            .none => return,
```

Please leave this particular hunk out; see #21306.

File: lib/std/c.zig [hunk 3]
```diff
@@ -8723,7 +8723,7 @@ pub const NOTE = switch (native_os) {
         pub const EXIT_DETAIL = 0x02000000;
         /// mask for signal & exit status
         pub const PDATAMASK = 0x000fffff;
-        pub const PCTRLMASK = (~PDATAMASK);
+        pub const PCTRLMASK = ~@as(u32, PDATAMASK);
         pub const EXIT_DETAIL_MASK = 0x00070000;
```
Comment (alexrp) [inline lib/std/c.zig:8727]: From Apple's `event.h`:

```c
#define	NOTE_PCTRLMASK	0xf0000000		/* mask for hint bits */
#define	NOTE_PDATAMASK	0x000fffff		/* mask for pid */
```

I don't understand what the original intent was here. Likewise for the FreeBSD prong below.

File: lib/std/c/darwin.zig [hunk 4]
```diff
@@ -10,7 +10,7 @@ const mode_t = std.c.mode_t;
 const off_t = std.c.off_t;
 const pid_t = std.c.pid_t;
 const pthread_attr_t = std.c.pthread_attr_t;
-const sigset_t = std.c.segset_t;
+const sigset_t = std.c.sigset_t;
```
Comment (alexrp) [inline lib/std/c/darwin.zig:13]: This was fixed in #21581.

File: lib/std/c.zig [hunk 5]
```diff
@@ -8723,7 +8723,7 @@ pub const NOTE = switch (native_os) {
         pub const EXIT_DETAIL = 0x02000000;
         /// mask for signal & exit status
         pub const PDATAMASK = 0x000fffff;
-        pub const PCTRLMASK = (~PDATAMASK);
+        pub const PCTRLMASK = ~@as(u32, PDATAMASK);
```
Comment (alexrp) [inline lib/std/c.zig:8726]: From Apple's `event.h`:

```c
#define	NOTE_PCTRLMASK	0xf0000000		/* mask for hint bits */
#define	NOTE_PDATAMASK	0x000fffff		/* mask for pid */
```

I don't understand what the original intent was here; this just looks wrong. They should be defined to the values they have in the header. Likewise for the FreeBSD prong below.

**Comment by PauloCampana** [comment]
Pardon for the delay and maybe git problems
This branch should now only include fixes that don't conflict with other PRs and now with correct values for NOTE_PCTRLMASK

**Review by alexrp — APPROVED** [review]

**Comment by yhyadev** [comment]
> std.valgrind and std.os.windows fixes taken from #21271 and #21272, the author closed their PRs.

I forgot that the head repository in my GitHub account must not be deleted until all branches get merged.

**Pull request merged**